library/std/src/sys/windows/compat.rs

```diff
@@ -74,6 +75,20 @@ impl Module {
         NonNull::new(module).map(Self)
     }
 
+    /// Load the library (if not already loaded)
+    ///
+    /// # Safety
+    ///
+    /// The module must not be unloaded.
+    pub unsafe fn load_system_library(name: &CStr) -> Option<Self> {
+        let module = c::LoadLibraryExA(
```

Comment:
So previously we loaded this via GetModuleHandle no? Is there a reason to change? IIUC this one is UB (well, deadlock in practice) to call with the loader lock held, which is slightly unfortunate.

Reply:
(That said if theres a reason to use this instead, that's probably fine)

Reply:
I was looking into #78444 and yeah, it should properly be `LoadLibrary` because the only two dlls that are guaranteed loaded are ntdll and kernel32. But then again "api-ms-win-core-synch-l1-2-0" is a kind of a virtual dll that may resolve to kernel32 anyway (or might not).

My thought was it shouldn't be too much of an issue in practice considering that using synchronization while the loader lock is held is already very sketchy and that in practice the required module will (hopefully) already be loaded.

That said, I'd be ok with reverting this part if it's too controversial.

Reply:
Hm, fair enough, I think we can keep it as is and see if anybody ends up complaining. This should only be triggered (I think) if people actually need to block, anyway.